skrub/_interpolation_joiner.py

```diff
@@ -287,43 +291,46 @@ def transform(self, X):
             main_table, self._main_key, "'X' (the main table)"
         )
         key_values = self.vectorizer_.transform(
-            main_table[self._main_key].set_axis(self._aux_key, axis="columns")
+            sbd.set_column_names(
+                s.select(main_table, s.cols(*self._main_key)), self._aux_key
+            )
         )
         prediction_results = joblib.Parallel(self.n_jobs)(
             joblib.delayed(_predict)(
+                main_table,
                 key_values,
                 assignment["columns"],
                 assignment["estimator"],
                 propagate_exceptions=(self.on_estimator_failure == "raise"),
             )
             for assignment in self.estimators_
         )
-        prediction_results = self._check_prediction_results(prediction_results)
+        prediction_results = self._check_prediction_results(
+            main_table, prediction_results
+        )
         predictions = [res["predictions"] for res in prediction_results]
         predictions = [
             _join_utils.add_column_name_suffix(df, self.suffix) for df in predictions
         ]
-        for part in predictions:
-            part.index = main_table.index
-        return pd.concat([main_table] + predictions, axis=1)
+        return sbd.concat_horizontal(main_table, *predictions)
 
-    def _check_prediction_results(self, results):
+    def _check_prediction_results(self, main_table, results):
         checked_results = []
         failed_columns = []
         for res in results:
             new_res = dict(**res)
             if res["failed"]:
-                if set(res["columns"]).issubset(
-                    self.aux_table.select_dtypes("number").columns.values
-                ):
-                    dtype = float
-                else:
-                    dtype = object
-                pred = pd.DataFrame(
-                    columns=res["columns"],
-                    index=np.arange(res["shape"][0]),
-                    dtype=dtype,
-                )
+                _pred = [
+                    sbd.all_null_like(
+                        sbd.col(self.aux_table, col),
+                        length=res["shape"][0],
+                    )
+                    for col in res["columns"]
+                ]
+                _pred = [
```

Comment:
do we need to convert to dataframe and then concatenate? couldn't we just use `make_dataframe` and pass to it the list of columns in `_pred`?

also it is not usual to have local variable names starting with an underscore (except the special case of `_`)